### PR TITLE
Added IsGameOver()

### PR DESCRIPTION
**Pull request by FrankPudding**

### CODE REVIEW

File: QuadRelate.Helpers/BoardHelper.cs [hunk 1]
```diff
@@ -20,6 +20,78 @@ public static List<int> AvailableColumns(this Board board)
             return availableColumns;
         }
 
+        public static bool IsGameOver(this Board board)
+        {
+            // Check for horizontal wins
+            for (var y = 0; y < Board.Height; y++)
```
Comment (vmelia) [inline QuadRelate.Helpers/BoardHelper.cs:25]: This is a large method.
You have a comment: **// Check for horizontal wins.**
Perhaps that's a clue?
Do you need another hint?

File: QuadRelate.Tests/BoardHelperTests.cs [hunk 2]
```diff
@@ -83,5 +83,59 @@ public void AvailableColumns_ForPartlyFullBoard_ReturnsFullList()
...
+            Assert.False(_board.IsGameOver());
+        }
+
+        [Fact]
+        public void IsGameOver_ForHorizontalBottomLeft_ReturnsTrue()
+        {
+            // Place counters in the first four columns of the bottom row
+            for (var x = 0; x < 4; x++)
+            {
+                _board[x, 0] = Cell.Red;
```
Comment (vmelia) [inline QuadRelate.Tests/BoardHelperTests.cs]: In tests, you might be better off doing these as four lines:
E.g., 
_board[0, 0] = Cell.Red;
_board[1, 0] = Cell.Red; 
_board[2, 0] = Cell.Red; 
_board[3, 0] = Cell.Red; 

Production code gets more generic; tests get more specific.

File: QuadRelate.Helpers/BoardHelper.cs (hunk 1 above)
Comment (FrankPudding) [reply QuadRelate.Helpers/BoardHelper.cs:25]: That should be enough of a hint (I hope)

File: QuadRelate.Tests/BoardHelperTests.cs (hunk 2 above)
Comment (FrankPudding) [reply QuadRelate.Tests/BoardHelperTests.cs]: Added this in most recent commit

File: QuadRelate.Helpers/BoardHelper.cs (hunk 1 above)
Comment (FrankPudding) [reply QuadRelate.Helpers/BoardHelper.cs:25]: Split the method up into three smaller methods in the most recent commit

**Pull request merged**